docs/csharp/pattern-matching.md

```diff
@@ -2,31 +2,225 @@
+
+That code above is a classic expression of the *type pattern*: You're testing a variable
+to determine its type, and taking different action based on that type.
+
+This code becomes simpler using extensions to the `is` expression to assign
+a variable if the test succeeds:
+
+[!code-csharp[IsPatternExpression](../../samples/csharp/PatternMatching/PatternMatching/GeometricUtilities.cs#03_IsPatternExpression "is pattern expression")]
+
+In this updated version, `is` expression both tests the variable, and assigns
+it to a new variable of the proper type. Also, notice that this version includes
+the `Rectangle` type, which is a `struct`. The new `is` expression works with
+value types as well as reference types.
+
+Language rules for pattern matching expressions help you avoid misusing
+the results of a match expression. In the example above, the variables `s`,
+ `c`, and `r` are only in scope and definitely assigned when the respective
+pattern match expressions have `true` results. If you try and use either
+variable in another location, your code generates compiler errors.
+
+Let's examine both of those rules in detail, beginning with scope. The variable
+`c` is in scope only in the `else` branch of the first `if` statement. The variable
+`s` is in scope in the method `ComputeArea`. That's because each 
+branch of an `if` statement establishes a separate scope for variables. However, the `if` statement
+itself does not. That means variables declared in the `if` statement are in the
+same scope as the `if` statement (the method in this case.) This behavior is not
+specific to pattern matching, but is the defined behavior for variable scopes 
+and `if` and `else` statements.
+
+The variables `c` and `s` are assigned when the respective `if` statements are true
+because of the definitely assigned when true mechanism.
+
+> [!TIP]
+> The samples in this topic use the recommended construct where
+> a pattern match `is` expression definitely assigns the match
+> variable in the `true` branch of the `if` statement.
+> You could reverse the logic by saying `if (!(shape is Square s))`
+> and the variable `s` would be definitely assigned only in the
+> `false` branch. While this is valid C#, it is not recommended
+> because it is more confusing to follow the logic.
+
+These rules mean that you are unlikely to accidentally access the result
+of a pattern match expression when that pattern was not met.
+
+## Using pattern matching `switch` statements
+
+As time goes on, you may need to support other shape types. As the number
+of conditions you are testing grows, you'll find that using the `is` pattern
+matching expressions can become cumbersome. In addition to requiring `if`
+statements on each type you want to check, the `is` expressions are limited
+to testing if the input matches a single type.
+
+
+As the number of conditions grows, you'll find that the `switch` pattern
```

Comment:
This partly repeats a sentence in the previous paragraph. Best to just add this sentence to the end of the last paragraph, and say that "In this case, you'll find..."